mozilla-release/browser/base/content/aboutDialog.xul

```diff
@@ -44,8 +44,6 @@
       <vbox id="rightBox" flex="1">
         <label id="distribution" class="text-blurb"/>
         <label id="distributionId" class="text-blurb"/>
-#expand <label id="version">&version.basedon;__MOZ_APP_VERSION__</label>
```

Comment:
how about surrounding it with preprocessor rules, so it won't be added with next Firefox versions?
Other way to hide it would be to use CSS.